sets/ordered_set.go

```diff
@@ -0,0 +1,96 @@
+package sets
+
+import (
+	"encoding/json"
+	"sort"
+
+	"github.com/pkg/errors"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+type OrderedSet[T constraints.Ordered] map[T]struct{}
```

Comment:
It would be nice if `OrderedSet[T]` could be built on top of `Set[T]` without having all of the boilerplate below. I have a feeling that Go doesn't make this kind of inheritance easy, alas.

Reply:
Yea, agreed.  It wasn't obvious to me how to do that, although didn't look very hard.